### PR TITLE
Remove redundant pod helper

### PR DESCRIPTION
**Pull request by 0xmichalis**

### CODE REVIEW

File: pkg/controller/deployment/util/deployment_util.go [hunk 1]
```diff
@@ -786,31 +786,6 @@ func GetAvailableReplicaCountForReplicaSets(replicaSets []*extensions.ReplicaSet
...
-// for a deployment.
-func IsPodAvailable(pod *v1.Pod, minReadySeconds int32, now time.Time) bool {
-	if !controller.IsPodActive(pod) {
-		return false
-	}
-	// Check if we've passed minReadySeconds since LastTransitionTime
-	// If so, this pod is ready
-	for _, c := range pod.Status.Conditions {
-		// we only care about pod ready conditions
-		if c.Type == v1.PodReady && c.Status == v1.ConditionTrue {
```
Comment (janetkuo) [inline pkg/controller/deployment/util/deployment_util.go]: IIUC `v1.IsPodAvailable` doesn't check if `.Status == v1.ConditionTrue`. Should we add that check?

Comment (janetkuo) [inline pkg/controller/deployment/util/deployment_util.go]: nvmd, I see it's checked

**Review by janetkuo — APPROVED** [review]

**Comment by kargakis** [comment]
@k8s-bot unit test this

**Pull request merged**